src/Selenium2Library/keywords/_element.py

```diff
@@ -680,6 +679,31 @@ def remove_location_strategy(self, strategy_name):
         """
         self._element_finder.unregister(strategy_name)
 
+    def element_attribute_should_be(self, locator, attr):
+        """Checks whether element has attribute with value
+
+        Usage Example:
+        | Element Attribute Should Be | xpath=//input | key=value |
+        """
+        self._info("Checking attributes '%s' for element '%s'" % (
+                                                    attr, locator))
+        try:
+            key, value = attr.split("=", 1)
+        except ValueError:
+            raise AssertionError(
+                    "Attribute %s does not match pattern key=value" % attr)
+
+        el = self._element_find(locator, True, True)
+        retval = el.get_attribute(key)
+
+        if retval is None:
```

Comment:
I don't see a problem with testing for retval is None. That is the part of the contract of the `get_attribute` method - see [API docs](http://seleniumhq.github.io/selenium/docs/api/py/webdriver_remote/selenium.webdriver.remote.webelement.html#selenium.webdriver.remote.webelement.WebElement.get_attribute).
If you called this keyword like this:
`| Element Attribute Should Be | css=div | text=${None} |`
Then the method would receive u'text=None' as the value for `attr` and only pass if the attribute value was 'None'. RF would serialize ${None} into the string 'None'. If the key and value were separate arguments or as keyword arguments, then None could be passed as literally ${None}, but that is not how it was implemented.